qiskit_nature/mappers/second_quantization/logarithmic_mapper.py

```diff
@@ -0,0 +1,208 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Logarithmic Mapper."""
+
+import operator
+
+from fractions import Fraction
+from functools import reduce
+from typing import List, Union, Tuple
+
+import numpy as np
+
+from qiskit.opflow import PauliSumOp
+from qiskit.quantum_info.operators import SparsePauliOp, Operator
+from qiskit_nature.operators.second_quantization import SpinOp
+from .spin_mapper import SpinMapper
+
+
+class LogarithmicMapper(SpinMapper):
+    """A mapper for Logarithmic spin-to-qubit mapping [1].
+    In this local encoding transformation, each individual spin S system is represented via
+    the lowest lying 2S+1 states in a qubit system with the minimal number of qubits needed to
+    represent >= 2S+1 distinct states.
+
+    [1]: Mathis, S. V., Guglielmo, M., & Ivano, T. (2020).
+         Toward scalable simulations of lattice gauge theories on quantum computers.
+         Phys. Rev. D, 102 (9), 094501. 10.1103/PhysRevD.102.094501
+    """
+
+    def __init__(self, padding: float = 1, embed_upper: bool = True) -> None:
+        """
+        Args:
+            padding:
+                When embedding a matrix into the upper/lower diagonal block of a 2^num_qubits by
+                2^num_qubits matrix, pads the diagonal of the block matrix with
+                the value of `padding`.
+
+            embed_upper:
+                This parameter sets whether
+                the given matrix is embedded in the upper left hand corner or
+                the lower right hand corner of the larger matrix.
+                I.e. using embed_upper = True returns the matrix:
+                [[ matrix,    0             ],
+                [   0   , padding * I]]
```

Comment:
This will likely need to be wrapped in some sort of block in order for Sphinx to display this properly. Same below